php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -2505,46 +2504,13 @@ abstract class NDB_BVL_Instrument extends NDB_Page
                 }
             }
         } else if ($element['type'] === "date") {
-            if (get_class($this->form) === 'LorisForm') {
-                $element['options'] = [
-                    'mindate'         => "1990-01-01",
-                    'maxdate'         => "2000-12-31",
-                    'RequireResponse' => false,
-                ];
-                //$element['RequireResponse'] = false;
-            } else {
-                $html = new DOMDocument();
-                $html->loadHTML($element['html']);
-                // Parse the HTML to get the min and max date
-                $selects = $html->getElementsByTagName("select");
-                if ($selects->length > 0) {
-                    // Old school QuickForm style. This should be removed
-                    // after QuickForm is retired.
-                    $year = $selects->item(0);
-
-                    // If Item 0 is blank, item 1 is the minimum,
-                    $minYear = $year->childNodes->item(0)->textContent;
-
-                    // PHP's trim function doesn't support UTF-8's 0xC2A0 (&nbsp;),
-                    // which seems to be how loadHTML parses spaces. Trim whitespace
-                    // and then specifically trim that character before checking if
-                    // the year is empty, and if it's empty use the next option in
-                    // the dropdown (which should be non-empty)
-                    $minYear = trim($minYear);
-                    $minYear = trim($minYear, chr(0xC2).chr(0xA0));
-                    if (empty($minYear)) {
-                        $minYear = $year->childNodes->item(1)->textContent;
-                    }
-                    // the last item is the maximum year
-                    $maxYear = $year->lastChild->textContent;
-
-                    $element['options'] = [
-                        'mindate' => $minYear . "-01-01",
-                        'maxdate' => $maxYear . "-12-31",
-                    ];
+            $element['options'] = [
+                'mindate'         => "1990-01-01",
+                'maxdate'         => "2000-12-31",
+                'RequireResponse' => false,
+            ];
+        }
```

Comment:
why are these values even there ?

Reply:
I don't know, it's the body of the `if (get_class($this->form) === 'LorisForm') {}` part. I just removed the condition which changed the indentation.

Reply:
yeah i saw, should we just use the dateoptions already in the file ?
https://github.com/aces/Loris/blob/80b1dfa92d01f5ab5e2ab9918016758f51dbd487/php/libraries/NDB_BVL_Instrument.class.inc#L854

Reply:
seems like a good idea. Want me to change it or just create an issue? It'll likely make the PR much larger since I assume some tests depend on it

Reply:
hmm if it makes the PR alot larger forget it. it will be done eventually in a future check

Reply:
after the tests pass with the current state I'll try it and see how much bigger it makes it. I could be mistaken, it's just an assumption.

Reply:
Actually doing a `git grep` for those particular dates there seems to only be 2 tests that depend on it, I'll just fix it here

Reply:
It's not as simple as it looks because the dateOptions isn't populated (or isn't populated yet..) when the tests are called. I'll just create an issue.

Reply:
yeah its loaded in add metadata fields... we would have to move it to the constructor first.

what I'm more confused about is the fact that its not even checking if these options are already set, which means its overriding whatever is already working... ? how does that even work for projects ?

Reply:
Have you tried the instrument form API endpoint on a project? I think this function was added to generate the JSON for that, and I doubt it's used anywhere else.